src/Interpreters/InterpreterCreateQuery.cpp

```diff
@@ -957,6 +957,19 @@ BlockIO InterpreterCreateQuery::createTable(ASTCreateQuery & create)
     if (!created)   /// Table already exists
         return {};
 
+    /// Check type compatible for materialized dest table and select columns
+    if (create.select && create.is_materialized_view && create.to_table_id)
+    {
+        StoragePtr table = DatabaseCatalog::instance().getTable({create.database, create.table, create.uuid}, getContext());
+        const auto & output_columns = table->getInMemoryMetadataPtr()->getSampleBlock();
+        Block input_columns=InterpreterSelectWithUnionQuery(
+            create.select->clone(), getContext(),SelectQueryOptions().analyze()).getSampleBlock();
+        auto actions_dag = ActionsDAG::makeConvertingActions(
+            input_columns.getColumnsWithTypeAndName(),
+            output_columns.getColumnsWithTypeAndName(),
+            ActionsDAG::MatchColumnsMode::Position);
```

Comment:
```suggestion
            ActionsDAG::MatchColumnsMode::Name);
```

AFAIR materialized view matches the columns by name?

Reply:
ActionsDAG::MatchColumnsMode::Name can support:
```
CREATE TABLE src (n UInt64, s FixedString(16)) ENGINE=Memory;
CREATE TABLE dst (n UInt8, s String) ENGINE = Memory;
CREATE MATERIALIZED VIEW mv TO dst (n String) AS SELECT * FROM src;
```
And when use position , the third command will fail? right?
So in what kind of situation can we  use ActionsDAG::MatchColumnsMode::Position?

Reply:
`ActionsDAG::MatchColumnsMode::Position` is used in  queries with `UNION ALL`.